plugins/jobs/src/js/JobsRunNow.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-unused-vars */
+import React from "react";
+/* eslint-enable no-unused-vars */
+
+import { graphqlObservable } from "data-service";
+import gql from "graphql-tag";
+
+import { Subject } from "rxjs/Subject";
+import { defaultSchema } from "./JobsMenuClient";
+
+const runNowJobMutation = gql`
+  mutation {
+    runJob(jobId: $jobId) {
+      jobId
+    }
+  }
+`;
+
+let runNowJobGraphql = jobId => {
+  return graphqlObservable(runNowJobMutation, defaultSchema, jobId);
+};
+
+export const runNowEvent$ = new Subject();
+export const jobsRunNow$ = runNowEvent$.switchMap(jobId => {
+  return runNowJobGraphql(jobId);
+});
+
+export const jobsRunNowAction = (id, externalRunNowJobGraphql) => {
+  return {
+    label: "Run Now",
+    onItemSelect() {
+      if (externalRunNowJobGraphql) {
+        runNowJobGraphql = externalRunNowJobGraphql;
```

Comment:
This seems like a super hacky solution for testing to me. I would rather like to have the respective functions hoisted to achieve something like this.

Reply:
Yes, was frustrated and wanted to make this work. Definitely not optimal, I changed this.